src/com/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1159,9 +1159,10 @@
     (add-certname! certname))
   (let [timestamp (to-timestamp time)
         replaced  (sql/update-values :certnames
-                                     ["name=? AND (deactivated<? OR deactivated IS NULL)" certname timestamp]
-                                     {:deactivated nil})]
-    (pos? (first replaced))))
+                                     ["name=? AND deactivated<?" certname timestamp]
```

Comment:
I think, there is a subtlety here we need to be conscious of, while we always updated, we always held some sort of predicate lock during our transaction due to the repeatable read isolation level. Removing this I bet has some locking side-effect in that regard, and while it feels more correct may shake up something hidden we haven't seen before, probably only at scale as well.

Don't get me wrong, I see the benefit of this, its such a subtle find, I like it - but I'm wary. Maybe we can step through the logic at standup or something and see if we consider this a real problem or not. I can't help but think on the surface there isn't really a problem, since deactivated is always going to be null. I definitely think this might speed things up though however ...